Debounce/IBufferer.cs

```diff
@@ -0,0 +1,50 @@
+﻿// SPDX-FileCopyrightText: 2024 Frans van Dorsselaer
+//
+// SPDX-License-Identifier: MIT
+//
+// SPDX-FileContributor: Alain van den Berg
+
+namespace Dorssel.Utilities;
+
+/// <summary>
+/// This interface specifies the public API for the <see cref="Bufferer{TEvent}"/> class.
+/// </summary>
+public interface IBufferer<TEvent>
```

Comment:
Maybe we should abstract the event from the other APIs somehow, so `IDebounce` and `IBufferer`.

Also, maybe we should align the names. `IDebounce` is named after what it does (verb), `IBufferer` after what it is.

How about `IDebounceEvents` and `IBufferEvents` ?

Reply:
I was able to extract the common properties to IDebounceSettings. Could even extract it to a separate class and add to constructor, but didn't do that yet.
Also aligned the names as you suggested.

Not sure how to abstract the event. The issue is that IDebounceEvents is non-generic and the IBufferEvents is generic. Non-generic and generic interfaces for collections are also separate, e.g., https://learn.microsoft.com/en-us/dotnet/api/system.collections.ilist?view=net-8.0 and https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.ilist-1?view=net-8.0. We would need to make the Debouncer also trigger generic events, not sure if that is useful.

Reply:
I see how `Trigger` is different for both classes, and so is the Event. But I do like the common IDebounceSettings interface (that's what it does, for both).

Maybe the type-specific interfaces should be called `IDebouncer` and `IBufferer`, just like their implementations (both deriving from `IDebounceSettings`, of course, like you did). You have now renamed them `IDebounceEvents` and `IBufferEvents`, but I don't like those. Like you said in the other comment, each can very well debounce/buffer other things than events.

Reply:
On second thought, I don't like the names either. Don't know why I didn't push back right away :-)